### DISCUSSION TITLE
AggregateRoot::Debugging

### DISCUSSION THREAD

**Discussion by paneq** [RailsEventStore/rails_event_store#1157] [Ideas]
```
class Product
  include AggregateRoot
  include AggregateRoot::Debugging if Rails.env.development? # never use in production
```

`AggregateRoot::Debugging` could hook into `apply` and display (`puts`) and log (`Rails.logger.debug`) applied event and the state after applying. 

I believe it could be useful.

**Comment by andrzejkrzywda** [comment]
Potentially related - https://github.com/RailsEventStore/rails_event_store/issues/60
